boofuzz/sessions.py

```diff
@@ -1000,14 +1086,14 @@ def _pre_send(self, target):
             target (session.target): Target we are sending data to
         """
 
-        if len(self._pre_send_methods) > 0:
+        for monitor in target.monitors:
             try:
-                for f in self._pre_send_methods:
-                    self._fuzz_data_logger.open_test_step('Pre_Send callback: "{0}"'.format(f.__name__))
-                    f(target=target, fuzz_data_logger=self._fuzz_data_logger, session=self, sock=target)
+                self._fuzz_data_logger.open_test_step("Monitor {}.pre_send()".format(str(monitor)))
```

Comment:
This is the only place where we output a log message like this.
Either drop it here or add it to the other callbacks too. I'd say drop it and modify the log message inside the CallbackMonitor.

Reply:
What would you like to add inside CallbackMonitor? The Name of the function is already printed, and as it's a callback, I don't see much more information to print here.

Reply:
@mistressofjellyfish No desire here for a new message, it just seemed that this message (in Session) was redundant and could be removed, since CallbackMonitor has some robust logging already.

Although, now that I look at it, this message precedes the entire CallbackMonitor (or similar monitor) pre_send() step, whereas inside CallbackMonitor you have a single log message for each function call. So the messages aren't entirely identical.

Plus, you could have other monitors that do a pre_send step.

So, on second look it makes sense to me to keep as-is.